datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/validate_profile.py

```diff
@@ -0,0 +1,146 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+import json
+import os
+from os.path import isfile, join
+
+import click
+import jsonschema
+import yaml
+
+from ....constants import get_root
+from ...console import CONTEXT_SETTINGS, abort, echo_failure, echo_info, echo_success
+
+
+@click.command("validate-profile", short_help="Validate SNMP profiles", context_settings=CONTEXT_SETTINGS)
+@click.option('-f', '--file', help="Path to a profile file to validate")
+@click.option('-d', '--directory', help="Path to a directory of profiles to validate")
+@click.option('-v', '--verbose', help="Increase verbosity of error messages", is_flag=True)
+def validate_profile(file, directory, verbose):
+    profiles_list = find_profiles(file, directory)
+    contents = read_profile(profiles_list)
+    errors = validate_with_jsonschema(contents, verbose)
+    produce_errors(errors, verbose)
+
+
+class Profile:
+    def __init__(self):
+        self.file_path = ""
+        self.errors = []
+        self.invalid = False
+        self.contents = ""
+
+    def __repr__(self):
+        return self.file_path
+
+
+def find_profiles(file, directory):
+    profiles_list = []
+    profile = Profile()
+    if file:
+        profile.file_path = file
+        profiles_list.append(profile)
+        return profiles_list
+
+    if directory:
+        profiles_list = get_all_profiles_from_dir(directory)
+        return profiles_list
+
+    else:
+        profiles_path = os.path.join(get_root(), "snmp", "datadog_checks", "snmp", "data", "profiles")
+        profiles_list = [f for f in os.listdir(profiles_path) if isfile(join(profiles_path, f))]
+        profiles_path = os.path.join(get_root(), "snmp", "datadog_checks", "snmp", "data", "profiles")
+        profiles_list = get_all_profiles_from_dir(profiles_path)
```

Comment:
you might re-use the above call to `get_all_profiles_from_dir` using the `directory` variable